src/Microsoft.AspNet.Razor/CodeGenerators/Visitors/CSharpCodeVisitor.cs

```diff
@@ -174,50 +181,60 @@ protected override void Visit(DynamicCodeAttributeChunk chunk)
             var currentRenderingMode = Context.ExpressionRenderingMode;
             var currentTargetWriterName = Context.TargetWriterName;
 
-            Context.TargetWriterName = ValueWriterName;
+            if (!string.IsNullOrEmpty(currentTargetWriterName))
+            {
+                Writer.WriteStartMethodInvocation(Context.Host.GeneratedClassContext.WriteAttributeValueToMethodName)
+                    .Write(currentTargetWriterName)
+                    .WriteParameterSeparator();
+            }
+            else
+            {
+                Writer.WriteStartMethodInvocation(WriteAttributeValueMethodName);
+            }
 
-            Writer.WriteParameterSeparator()
-                   .WriteLine();
+            Context.TargetWriterName = ValueWriterName;
 
             var code = chunk.Children.FirstOrDefault();
             if (code is ExpressionChunk || code is ExpressionBlockChunk)
             {
-                Writer.WriteStartMethodInvocation("Tuple.Create")
-                        .WriteLocationTaggedString(chunk.Prefix)
-                        .WriteParameterSeparator()
-                        .WriteStartMethodInvocation("Tuple.Create", new string[] { "System.Object", "System.Int32" });
+                Writer
+                    .WriteLocationTaggedString(chunk.Prefix)
+                    .WriteParameterSeparator();
 
                 Context.ExpressionRenderingMode = ExpressionRenderingMode.InjectCode;
 
                 Accept(code);
 
-                Writer.WriteParameterSeparator()
-                       .Write(chunk.Start.AbsoluteIndex.ToString(CultureInfo.CurrentCulture))
-                       .WriteEndMethodInvocation(endLine: false)
-                       .WriteParameterSeparator()
-                       .WriteBooleanLiteral(value: false)
-                       .WriteEndMethodInvocation(endLine: false);
+                Writer
+                    .WriteParameterSeparator()
+                    .Write(chunk.Start.AbsoluteIndex.ToString(CultureInfo.InvariantCulture))
+                    .WriteParameterSeparator()
+                    .Write(chunk.Association.Length.ToString(CultureInfo.InvariantCulture))
```

Comment:
Really starting to think you're mixing an unrelated change into this PR. If there's a culture-related issue, file it separately.